alicloud/resource_alicloud_route_table_association.go

```diff
@@ -0,0 +1,135 @@
+				ForceNew: true,
+			},
+
+			"vswitch_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceAliyunRouteTableAssociationCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AliyunClient)
+
+	args := vpc.CreateAssociateRouteTableRequest()
+	args.RouteTableId = Trim(d.Get("route_table_id").(string))
+	args.VSwitchId = Trim(d.Get("vswitch_id").(string))
+	if err := resource.Retry(3*time.Minute, func() *resource.RetryError {
+		ar := args
+		if _, err := client.vpcconn.AssociateRouteTable(ar); err != nil {
+			if IsExceptedError(err, TaskConflict) {
+				return resource.RetryableError(fmt.Errorf("AssociateRouteTable got an error: %#v", err))
+			}
+			return resource.NonRetryableError(fmt.Errorf("AssociateRouteTable got an error: %#v", err))
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+	//Waiting for the route entrys
+	time.Sleep(3 * time.Second)
+	if err := client.WaitForAllRouteEntries(args.RouteTableId, Available, DefaultTimeout); err != nil {
+		return fmt.Errorf("Wait for route entry got error: %#v", err)
+	}
+	d.SetId(args.RouteTableId + ":" + args.VSwitchId)
+
+	return resourceAliyunRouteTableAssociationRead(d, meta)
+}
+
+func resourceAliyunRouteTableAssociationRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AliyunClient)
+
+	routeTableId, vSwitchId, err := getRouteTableIdAndVSwitchId(d, meta)
+	if err != nil {
+		return err
+	}
+
+	routeTable, err := client.DescribeRouteTable(routeTableId)
+
+	if err != nil {
+		if NotFoundError(err) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error Describe Route Table Attribute: %#v", err)
+	}
+
+	//Finding the vSwitchId
+	if len(routeTable.VSwitchIds.VSwitchId) <= 0 {
+		d.SetId("")
+		return nil
+	}
+	flag := -1
+	length := len(routeTable.VSwitchIds.VSwitchId)
+	for i := 0; i < length; i++ {
+		if routeTable.VSwitchIds.VSwitchId[i] == vSwitchId {
+			flag = i
+		}
+	}
+	if flag == -1 {
+		return GetNotFoundErrorFromString(GetNotFoundMessage("VSwitchId", routeTableId))
+	}
+
+	d.Set("route_table_id", routeTable.RouteTableId)
+	d.Set("vswitch_id", vSwitchId)
+	return nil
+}
+
+func resourceAliyunRouteTableAssociationDelete(d *schema.ResourceData, meta interface{}) error {
+
+	client := meta.(*AliyunClient)
+
+	routeTableId, vSwitchId, err := getRouteTableIdAndVSwitchId(d, meta)
+	if err != nil {
+		return err
+	}
+
+	request := vpc.CreateUnassociateRouteTableRequest()
+	request.RouteTableId = routeTableId
+	request.VSwitchId = vSwitchId
+
+	return resource.Retry(5*time.Minute, func() *resource.RetryError {
+		_, err := client.vpcconn.UnassociateRouteTable(request)
+		//Waiting for unassociate the route table
+		time.Sleep(10 * time.Second)
+		if err != nil {
+			if IsExceptedError(err, TaskConflict) {
+				return resource.RetryableError(fmt.Errorf("Unassociate Route Table timeout and got an error:%#v.", err))
+			}
+		}
+		return nil
```

Comment:
Please add one operation to ensure the vsw has been unassociated truly after running `UnassociateRouteTable`.